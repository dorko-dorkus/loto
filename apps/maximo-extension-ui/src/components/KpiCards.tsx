import React from 'react';

export interface KpiItem {
  label: string;
  value: string | number;
}

interface KpiCardsProps {
  items: KpiItem[];
}

export default function KpiCards({ items }: KpiCardsProps) {
  return (
    <div className="mb-4 grid grid-cols-1 gap-4 sm:grid-cols-3">
      {items.map(({ label, value }) => (
        <div
          key={label}
          className="rounded-[var(--mxc-radius-md)] border border-[var(--mxc-border)] p-4 text-center shadow-[var(--mxc-shadow-sm)]"
        >
          <div className="text-2xl font-semibold">{value}</div>
          <div className="mt-2 text-sm">{label}</div>
        </div>
      ))}
    </div>
  );
}

