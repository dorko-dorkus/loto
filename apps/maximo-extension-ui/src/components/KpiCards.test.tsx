import { render, screen } from '@testing-library/react';
import { test, expect } from 'vitest';
import KpiCards from './KpiCards';

const kpis = [
  { label: 'Active', value: 1 },
  { label: 'Completed', value: 2 }
];

test('renders KPI labels', () => {
  render(<KpiCards items={kpis} />);
  expect(screen.getByText('Active')).toBeInTheDocument();
  expect(screen.getByText('Completed')).toBeInTheDocument();
});

