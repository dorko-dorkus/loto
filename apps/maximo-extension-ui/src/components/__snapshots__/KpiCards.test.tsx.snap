// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  class="mb-4 grid grid-cols-1 gap-4 sm:grid-cols-3"
>
  <div
    class="rounded-[var(--mxc-radius-md)] border border-[var(--mxc-border)] p-4 text-center shadow-[var(--mxc-shadow-sm)]"
  >
    <div
      class="text-2xl font-semibold"
    >
      1
    </div>
    <div
      class="mt-2 text-sm"
    >
      Active
    </div>
  </div>
  <div
    class="rounded-[var(--mxc-radius-md)] border border-[var(--mxc-border)] p-4 text-center shadow-[var(--mxc-shadow-sm)]"
  >
    <div
      class="text-2xl font-semibold"
    >
      2
    </div>
    <div
      class="mt-2 text-sm"
    >
      Completed
    </div>
  </div>
</div>
`;
